include/cuco/sketch_size.cuh

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+namespace cuco {
+
+/**
+ * @brief Strng type for specifying the sketch size of cuco::distinct_count_estimator(_ref) in KB.
+ *
+ * Values can also be given as literals, e.g., 64.3_KB
+ */
+class sketch_size_kb {
+ public:
+  /**
+   * @brief Constructs a sketch_size_kb object.
+   *
+   * @param value The size of a sketch given in KB
+   */
+  __host__ __device__ explicit constexpr sketch_size_kb(double value) noexcept : value_{value} {}
+
+  /**
+   * @brief Conversion to value type.
+   *
+   * @return Sketch size in KB
+   */
+  __host__ __device__ constexpr operator double() const noexcept { return this->value_; }
+
+ private:
+  double value_;  ///< Sketch size in KB
+};
```

Comment:
Good catch regarding the `strong_type` typedef. Both `sketch_size_kb` and `standard_deviation` are not only used in the public `distinct_count_estimator(_ref)' classes, but also in the implementation classes `hyperloglog(_ref)`. So, if we want to get rid of the separate headers, we need to move them to lowest denominator which is `hyperloglog_ref.cuh`. The problem with that is that users need to consult a `detail` header for the inline documentation of said types.

Reply:
Oh well, if I define both `sketch_size_kb` and `standard_deviation` as type aliases for `strong_type::double` I get some declaration errors:

```
/workspaces/cuCollections/include/cuco/detail/hyperloglog/hyperloglog_ref.cuh(489): error: invalid redeclaration of member function "cuco::detail::hyperloglog_ref<T, Scope, Hash>::sketch_bytes(cuco::sketch_size_kb) noexcept" (declared at line 474)
                                     static constexpr std::size_t sketch_bytes(
```

If I declare the typedef for only one of them everything works fine. It seems that the compiler thinks they are both the same type - not what we want from a named strong type argument.

Reply:
> It seems that the compiler thinks they are both the same type 

:face_exhaling: 

Can we use `double` instead of the `standard_deviation` strong type? I still find having two public headers to wrap the hll-related types to be too much.

Reply:
https://www.fluentcpp.com/2016/12/08/strong-types-for-strong-interfaces/

The section `Using phantoms to be stronger` may help here

Reply:
Thanks for the pointer! This unfortunately means we have to update all occurrences of `strong_type` in our code base I think.

> Can we use double instead of the standard_deviation strong type? I still find having two public headers to wrap the hll-related types to be too much.

Using a plain `double` would defy the purpose of having the type-safetiness of strong types, especially since the other overload takes a `sketch_size_kb` type which is implicitly convertible to a plain double. This is a recipe for trouble. We could place both types into a single file (how should we name it?) or we place all of the plain strong types into a separate directory as I proposed earlier.

Reply:
#451 fixes the problem with the indistinguishable strong types.
With this PR, defining a new strong type can be done in a single line.
To reduce the number of headers we could place all of the strong type definitions in cuco, i.e., sentinels, `standard_deviation', 'sketch_size_kb`, into a single file, e.g., `include/cuco/strong_types.cuh`

Reply:
Based on offline discussion, I moved all strong type definitions to `include/cuco/types.cuh` in [9b0ee68](https://github.com/NVIDIA/cuCollections/pull/429/commits/9b0ee68e6eabc60015d3f7746e109b80989e7a50)